manifests/config.pp

```diff
@@ -7,28 +7,36 @@
     group => $dns::params::group,
     mode  => '0640',
   }
+
   concat::fragment { 'dns_zones+01-header.dns':
     target  => $::dns::publicviewpath,
     content => ' ',
     order   => '01',
   }
 
-  file {
-    $dns::namedconf_path:
-      owner   => root,
-      group   => $dns::params::group,
-      mode    => '0640',
-      content => template($::dns::namedconf_template);
-    $dns::optionspath:
-      owner   => root,
-      group   => $dns::params::group,
-      mode    => '0640',
-      content => template($::dns::optionsconf_template);
-    $dns::zonefilepath:
-      ensure  => directory,
-      owner   => $dns::params::user,
-      group   => $dns::params::group,
-      mode    => '0640';
+  concat { [$::dns::namedconf_path, $::dns::optionspath]:
+    owner => root,
+    group => $::dns::params::group,
+    mode  => '0640',
+  }
+
+  concat::fragment { 'namedconf_template':
```

Comment:
Use the existing naming schema for fragments, i.e. `named.conf+10-main.dns` or similar.